angular/projects/admin/src/app/admin/form/date-time/date-time.component.ts

```diff
@@ -0,0 +1,144 @@
+import { firestore } from 'firebase/app';
+
+@Component({
+  selector: 'tanam-date-time',
+  templateUrl: './date-time.component.html',
+  styleUrls: ['./date-time.component.scss'],
+  providers: [
+    {
+      provide: MatFormFieldControl,
+      useExisting: DateTimeComponent,
+    },
+  ],
+})
+export class DateTimeComponent implements MatFormFieldControl<firestore.Timestamp>, ControlValueAccessor, OnInit, OnDestroy {
+
+  private static _nextId = 0;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+  @HostBinding() id = `date-time-${DateTimeComponent._nextId++}`;
+  @ViewChild('container', { read: ElementRef }) container: ElementRef;
+
+
+  @Input() type: string;
+  @Input() title: string;
+  @Input() key: string;
+
+
+  stateChanges = new Subject<void>();
+  controlType = 'date-time';
+  shouldLabelFloat = false;
+
+  required: boolean;
+  errorState: boolean;
+  autofilled?: boolean;
+  editorData: any;
+
+  private _disabled = false;
+  private _focused = false;
+  private _placeholder: string;
+  private _onTouchedCallback: () => void;
+  private _onChangeCallback: (value: any) => void;
+
+  constructor(
+    @Optional() @Self() public ngControl: NgControl,
+    private readonly focusMonitor: FocusMonitor,
+    private readonly elementRef: ElementRef<HTMLElement>,
+  ) {
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+
+    focusMonitor.monitor(elementRef.nativeElement, true).subscribe(origin => {
+      this.focused = !!origin;
+      this.stateChanges.next();
+    });
+  }
+
+  @Input()
+  get placeholder() {
+    return this._placeholder;
+  }
+  set placeholder(placeholder) {
+    this._placeholder = placeholder;
+    this.stateChanges.next();
+  }
+
+  get empty() {
+    return !this.editorData || this.editorData.trim().length === 0;
+  }
+
+  @Input()
+  get disabled() {
+    return this._disabled;
+  }
+  set disabled(flag) {
+    this._disabled = flag;
+    this.stateChanges.next();
+  }
+
+  @Input()
+  get value(): firestore.Timestamp {
+    return this.editorData;
+  }
+  set value(value: firestore.Timestamp) {
+    this.editorData = value;
+    this.stateChanges.next();
+  }
+
+  get focused() {
+    return this._focused;
+  }
+
+  set focused(flag: boolean) {
+    this._focused = coerceBooleanProperty(flag);
+    this._onTouchedCallback();
+  }
+
+  ngOnInit() {
+  }
+
+  inputChange($event: moment.MomentInput ) {
+    const toDate = moment($event).toDate();
+    const toTimestamp = firestore.Timestamp.fromDate(toDate);
+    this._onChangeCallback(toTimestamp);
+    this.stateChanges.next();
+  }
+
+  ngOnDestroy() {
+    this.stateChanges.complete();
+    this.focusMonitor.stopMonitoring(this.elementRef.nativeElement);
+  }
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  onContainerClick(event: MouseEvent): void {
+    this._onTouchedCallback();
+  }
+
+  writeValue(text: firestore.Timestamp): void {
```

Comment:
Parameter name should be `timestamp` not `text`

Reply:
Resolve on this commit https://github.com/oddbit/tanam/pull/56/commits/afe64bb0f806d9ac25dc11b55c4a023e3c778a9d